app/src/main/java/com/io/moviO/movieDetails/MovieDetailsFragment.kt

```diff
@@ -21,17 +23,18 @@ class MovieDetailsFragment : Fragment(R.layout.fragment_movie_details) {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         binding = FragmentMovieDetailsBinding.bind(view)
-        requireArguments().getString(ARG_MOVIE_ID)?.let { viewModel.getMovieById(it) }
+        requireArguments().getInt(ARG_MOVIE_ID).let { viewModel.getMovieById(it) }
         viewModel.movie.observe(viewLifecycleOwner, Observer {
             when (it) {
                 is DataResult.Success ->
                     binding.apply {
-                        movieName.text = it.value.name
-                        movieYear.text = it.value.year
-                        movieGenre.text = it.value.gerne
-                        movieCast.text = it.value.cast
+                        movieName.text = it.value.title
+                        movieReleaseDate.text = it.value.releaseDate
+                        movieGenre.text = convertGenres(it.value.genres!!)
+                        movieRating.text = it.value.voteAverage.toString().plus("/10")
                         overviewText.text = it.value.overview
-                        Glide.with(this@MovieDetailsFragment).load(it.value.poster)
+                        Glide.with(this@MovieDetailsFragment)
+                            .load("${Constants.IMAGE_URL_START_PART}${it.value.imageUrl}")
```

Comment:
figure out a way for presentation (Fragment, Adapter etc) to not know about `Constants.IMAGE_URL_START_PART` they shouldn't be bothered to know where this image comes from.

They should just load it.value.imageUrl into a view they have, no idea where that images comes from

Reply:
resolved